tests/utils.py

```
@@ -0,0 +1,241 @@
+    """
+    colors = 0
+    stdin_isatty = True,
+    stdout_isatty = True
+    is_windows = False
+
+    _shutil_rmtree = shutil.rmtree  # needed by __del__ (would get gc'd)
+
+    def __init__(self, **kwargs):
+
+        if 'stdout' not in kwargs:
+            kwargs['stdout'] = tempfile.TemporaryFile('w+b')
```

Ruff (0.12.2)

Use a context manager for opening files

(SIM115)